### PR TITLE
add blog post for the new wireguard mode

### PR DESCRIPTION
**Pull request by decathorpe**
I have never used hugo before, so I'm not sure that everything is 100% the way it needs to be (though it's reasonably similar to Jekyll, which I *have* used).

Feel free to modify / shorten / extend / adapt the text itself, if necessary. 🙃

### CODE REVIEW

File: src/content/posts/new-wireguard-mode/index.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    name: Fabio Valentini
+    twitter: decathorpe
+---
+
+The next release of mitmproxy is slated to ship with a new alternative to the existing `transparent` mode, based on
+WireGuard. The new `wireguard` mode was designed to be much simpler to set up and use, and in most cases, it does not
+require taking any actions with elevated privileges.
+
+Connecting another device and routing its traffic through mitmproxy becomes as simple as connecting to a WireGuard VPN
+--- either by using the WireGuard configuration that is printed by mitmproxy, or by scanning the QR code that is shown
```
Comment (Kriechi) [inline src/content/posts/new-wireguard-mode/index.md:20]:
```suggestion
--- either by using the WireGuard configuration file generated by mitmproxy, or by scanning the QR code that is shown
```

File: src/content/posts/new-wireguard-mode/index.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+in the mitmweb interface.
+
+## How it works
+
+The core of this feature is implemented in the [`mitmproxy-wireguard`](https://pypi.org/project/mitmproxy-wireguard/)
```
Comment (Kriechi) [inline src/content/posts/new-wireguard-mode/index.md:25]: will the main mitmproxy package pull in this dependency automatically (for all users), or do interested users need to install mitmproxy-wireguard explicitly?

File: src/content/posts/new-wireguard-mode/index.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    twitter: decathorpe
+---
+
+The next release of mitmproxy is slated to ship with a new alternative to the existing `transparent` mode, based on
+WireGuard. The new `wireguard` mode was designed to be much simpler to set up and use, and in most cases, it does not
+require taking any actions with elevated privileges.
+
+Connecting another device and routing its traffic through mitmproxy becomes as simple as connecting to a WireGuard VPN
+--- either by using the WireGuard configuration that is printed by mitmproxy, or by scanning the QR code that is shown
+in the mitmweb interface.
```
Comment (Kriechi) [inline src/content/posts/new-wireguard-mode/index.md:21]: A screenshot would be great! Or even a 30sec gif/video from a phone scanning the QR code from mitmweb and it auto-configured the wireguard phone app.

File: src/content/posts/new-wireguard-mode/index.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+&nbsp;
+
+&nbsp;
+
+&nbsp;
+
+&nbsp;
+
+&nbsp;
```
Comment (Kriechi) [inline src/content/posts/new-wireguard-mode/index.md:88]: Please remove these blank paragraphs.

File: src/content/posts/new-wireguard-mode/index.md (hunk 3 above)
Comment (mhils) [reply src/content/posts/new-wireguard-mode/index.md:21]: Video I recorded the other day:
https://user-images.githubusercontent.com/1019198/192087170-9bf30c93-a2cd-4c24-8075-78963f4bb567.mp4

Screenshot:
image

File: src/content/posts/new-wireguard-mode/index.md (hunk 1 above)
Comment (decathorpe) [reply src/content/posts/new-wireguard-mode/index.md:20]: I'll change this with the next push, thanks.

File: src/content/posts/new-wireguard-mode/index.md (hunk 3 above)
Comment (decathorpe) [reply src/content/posts/new-wireguard-mode/index.md:21]: Great, I'll try to embed these two files :)

File: src/content/posts/new-wireguard-mode/index.md (hunk 4 above)
Comment (decathorpe) [reply src/content/posts/new-wireguard-mode/index.md:88]: I copied these from an existing blog post, but sure, I'll remove them.

File: src/content/posts/new-wireguard-mode/index.md (hunk 2 above)
Comment (decathorpe) [reply src/content/posts/new-wireguard-mode/index.md:31]: It's a new dependency and will be included by default. I added a sentence to clarify this.

**Comment by decathorpe** [comment]
I think I addressed all comments with the new version, please take another look.
I'm not sure if I embedded the video correctly, though, I don't know hugo syntax, I tried to base it off of the hugo docs.

**Pull request merged**